lib/ecto/adapters/sql.ex

```diff
@@ -596,6 +602,23 @@ defmodule Ecto.Adapters.SQL do
     end
   end
 
+  ## Migrations
+
+  @doc false
+  def lock_for_migrations(repo, table, opts, fun) do
+    import Ecto.Query, only: [from: 2]
+
+    transaction(repo, opts, fn ->
+      versions =
+        from(p in table, select: type(p.version, :integer))
+        |> Map.put(:prefix, opts[:prefix])
+        |> Map.put(:lock, "FOR UPDATE")
+        |> repo.all()
```

Comment:
What happens if `repo.all` fails? How will this error cascade and show up to users? You can for example use "FOR WHATEVER" as the lock and see how it will behave.

Reply:
To clarify, an error can happen if the user already has a schema_migrations table but it doesn't have a version field or something of sorts.